src/structs/neighbor_summary.rs

```diff
@@ -8,7 +8,9 @@ use std::num::Wrapping;
 
 #[derive(Copy, Clone)]
 pub struct NeighborSummary {
-    edge_pixels: [i16; 16],
+    edge_pixels_h: [i16; 8],
+
+    edge_pixels_v: [i16; 8],
```

Comment:
interesting that breaking this into two structs allows us to prove to the compiler that they don't alias.
Theoretically split_at_mut might be able to do similar things, but I do enjoy the fixed bound arrays as opposed to slices--it certainly makes it easier for the type system to reason about things.